content/articles/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/index.md

```diff
@@ -0,0 +1,952 @@
+
+         //reset the fields
+         this.todo.title = "";
+         this.todo.description = "";
+         
+         // set the message
+         this.todo.message = "Todo edited successfully";
+
+         return;
+         }catch(error){
+             this.todo.error = error;
+         return;
+         }
+     }
+     this.todo.error = null;
+     if(!this.todo.title){
+         this.todo.error = "Title is required";
+         return;
+     }
+     if(!this.todo.description){
+         this.todo.error = "Description is required";
+         return;
+     }
+     e.preventDefault();
+     },
+
+ },
+ created(){
+     // Called on load
+     this.getTodo()
+ },
+ };
+ </script>
+```
+
+From above we are, exporting the todo data, getting the todo when the page is loaded, handling custom validation, and data submission of the edited todo.
+
+- After setting up the components, we need to handle the routing into various pages. To do this, we will add the following in the `src/main.js` file:
+
+```javascript
+// create a vuerouter instance
+const router = new VueRouter({
+mode: 'history',
+base: __dirname,
+routes: [
+    { path: '/', component: Todos,name:'home' },
+    { path: '/add-todo', component: AddTodo,name:'add-todo' },
+    { path: '/edit-todo/:id', component: EditTodo,name:'edit-todo' },
+]
+});
+
+// pass the router to the app config
+new Vue({
+router:router,
+render: h => h(App),
+}).$mount('#app');
+```
+
+From above, we are creating a `VueRouter` instance passing in the `mode`, `base`, and `routes`. For the routes, we pass the `path`, `component`, and `name` for each.
+
+After creating the instance, we pass it to the Vue object.
+
+With that, we are ready to start the development server and test the functionalities we have implemented.
+
+To do that, run the following command:
+
+```bash
+npm run serve
+```
+
+The above command will spin up the development server on port `8080`. You can access your app from `http://localhost:8080`.
+
+Your application should resemble the following:
+
+*Todos page*
+
+![todos-page](/engineering-education/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/todos-page.png)
+
+*Add todo page*
+
+![add-todo-page](/engineering-education/how-to-build-a-vue-app-with-flask-sqlite-backend-using-docker/add-todos-page.png)
+
+### Dockerizing the application
+To dockerize the application we have built, we will follow the following steps:
+
+#### Dockerize the Flask API
+To dockerize the API, create a Dockerfile, and a `.dockerignore` file in the API folder.
+
+The Dockerfile will host the instructions for creating the image, whereas the `.dockerignore` file will host the files to be ignored when copying to the image.
+
+Add the following in the Dockerfile:
+
+```dockerfile
+# Base python package
+FROM python:3.8-slim-buster
+
+# Working directory
+WORKDIR /app
+
+# Copy the dependencies
+COPY requirements.txt requirements.txt
+
+# Install the dependencies
+RUN pip3 install -r requirements.txt
+
```

Comment:
You are working with `requirements.txt` but you haven't generated this file yet. Please fix this.